sdk/file_sdk/Common/file_format_base.h

```diff
@@ -0,0 +1,36 @@
+#ifndef FILE_FORMAT_BASE_H
+#define FILE_FORMAT_BASE_H
+
+#include <string>
+#include <vector>
+#include "../../rmscrypto_sdk/CryptoAPI/IStream.h"
+#include "tag.h"
+using std::string;
+using std::vector;
+
+namespace mip{
+namespace file {
+
+class IFileFormat
+{
+public:
+  IFileFormat(rmscrypto::api::SharedStream& file, string extension);
+
+  virtual string GetOriginalExtension() = 0;
+
+  virtual void SetTags(vector<Tag>& tags);

```

Comment:
+1. 
Make this class as pure interface without members.
And implement FileFormatBase class that inherits from this interface

---
In reply to: [134246608](https://github.com/AzureAD/rms-sdk-for-cpp/pull/133#discussion_r134246608) [](ancestors = 134246608)